src/main/java/com/groupseven/sysc4806project/ShoppingCartService.java

```diff
@@ -0,0 +1,109 @@
+package com.groupseven.sysc4806project;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.*;
+
+@RestController
+@RequestMapping("/api/carts")
+public class ShoppingCartService {
+
+
+    private final ShoppingCartRepository shoppingCartRepository;
+    @Autowired
+    private BookRepository bookRepository;
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    public ShoppingCartService (ShoppingCartRepository shoppingCartRepository){
+        this.shoppingCartRepository = shoppingCartRepository;
+    }
+
+    @GetMapping("/{cartId}")
+    public ShoppingCart getCart(@PathVariable int cartId){
+        return shoppingCartRepository.findById(cartId).orElse(null);
+    }
+
+    @GetMapping("/create/{userId}")
+    public Integer createCart(@PathVariable int userId){
+        User user = userRepository.findById(userId).orElse(null);
+        assert user != null;
+        ShoppingCart cart = new ShoppingCart();
+        cart.setCustomer(user);
+        shoppingCartRepository.save(cart);
+        return cart.getId();
+    }
+
+    @GetMapping("/{cartId}/books")
+    public List<Book> listBooksInCart(@PathVariable int cartId){
+        ShoppingCart cart = shoppingCartRepository.findById(cartId).orElse(null);
+        assert cart != null;
```

Comment:
Same thing with assert comment above. Might make sense to return null or an empty list here